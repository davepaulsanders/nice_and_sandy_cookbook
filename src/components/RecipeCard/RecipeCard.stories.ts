import type { Meta, StoryObj } from '@storybook/react';
 
import RecipeCard from './RecipeCard';
 
const meta: Meta<typeof RecipeCard> = {
  component: RecipeCard,
};
 
export default meta;
type Story = StoryObj<typeof RecipeCard>;
export const RecipeCards: Story = {
	args: {
	img: "https://www.feastingathome.com/wp-content/uploads/2024/10/Vegan-Fajitas-13.jpg",
	href: "https://www.feastingathome.com/vegan-fajitas/",
	label: "Vegan Fajitas",
	alt: "test",
	}
};
