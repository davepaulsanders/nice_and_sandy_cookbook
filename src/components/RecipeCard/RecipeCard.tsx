import CardHeader from "../CardHeader/CardHeader"
import CardContent from "../CardContent/CardContent"
import tack from "/thumb-tack.webp" 
import cancelPin from "/thumb-tack-cancel.webp" 

const RecipeCard = (props) => {
	const pinRecipeToggle = () => {
		if (props.isPinned) {
			props.setPinnedRecipes(prev => prev.filter(recipe => recipe.props.id !== props.id))
		} else {
			props.setPinnedRecipes(prev => [<RecipeCard isPinned={true} setPinnedRecipes={props.setPinnedRecipes} img={props.img} href={props.href} label={props.label} alt={props.alt}/>, ...prev])
	
		}
	}
	return (
		<div className="relative bg-lightg flex flex-col justify-content-center \
		items-center shadow-lg rounded-md">
			<CardHeader {...props} />
			<CardContent {...props} />
			<img onClick={pinRecipeToggle} width="30px" height="30px" className="absolute top-5 right-5 hover:scale-125 hover:cursor-pointer" src={props.isPinned ? cancelPin : tack}/>
		</div>
	)
	}
export default RecipeCard
