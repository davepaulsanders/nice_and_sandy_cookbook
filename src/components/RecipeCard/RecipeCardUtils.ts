import { fetchData } from "../../utils/fetch"

export const pinRecipeToggle = async (e, props) => {
		e.preventDefault()
		const {recipes, setRecipes, is_pinned, ...rest} = props
		await fetchData(`../v1/recipes/pinned/${props.id}`, 
					{
						method: "PATCH",
						"Content-Type": "application/json"
				})
		if (props.is_pinned) {
			props.setRecipes(prev => {
				return prev.map((rec) => {
				if (rec.id === props.id) {
				  return { ...rest, is_pinned: false};
				}
				return rec;
				})
			  })

		} else {
			props.setRecipes(prev => {
				return prev.map((rec) => {
				if (rec.id === props.id) {
				  return { ...rest, is_pinned: true};
				}
				return rec;
				})
			  })
		}
	}
