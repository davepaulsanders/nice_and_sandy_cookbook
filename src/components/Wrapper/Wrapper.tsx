const Wrapper = ({children}: ReactNode[]) => {
	return (
		<div className="grid gap-x-8 grid-cols-12 grid-rows-auto w-11/12 sm:mx-auto">
		{children.map((child, i) => (<div key={`c${child.props.alt}${i}`} className="col-span-12 sm:col-span-3">{child}</div>))}
		</div>
	)
}
export default Wrapper
	
