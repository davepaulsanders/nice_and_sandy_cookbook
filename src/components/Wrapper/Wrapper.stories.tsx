import { ReactElement } from "react"
import type { Meta, StoryObj } from '@storybook/react';
 
import Wrapper from './Wrapper';
import RecipeCard from '../RecipeCard/RecipeCard';
 
const meta: Meta<typeof Wrapper> = {
  component: Wrapper,
};


const children: ReactElement[] = []
for (let i = 0; i < 10; i++) {
	children.push(<RecipeCard 
	img="https://www.feastingathome.com/wp-content/uploads/2024/10/Vegan-Fajitas-13.jpg"
	href="https://www.feastingathome.com/vegan-fajitas/"
	label="Vegan Fajitas"
	alt="test" />)
	}
 
export default meta;
type Story = StoryObj<typeof Wrapper>;
export const WrapperStory: Story = {args: {
	children
}}
