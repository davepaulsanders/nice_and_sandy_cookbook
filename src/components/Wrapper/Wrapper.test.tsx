import Wrapper from "./Wrapper" 
import RecipeCard from "../RecipeCard/RecipeCard" 
import { expect, test } from 'vitest'
import { screen, render } from '@testing-library/react'
import '@testing-library/dom'

const children: ReactNode[] = []
for (let i = 0; i < 10; i++) {
	children.push(<RecipeCard 
	img="https://www.feastingathome.com/wp-content/uploads/2024/10/Vegan-Fajitas-13.jpg"
	href="https://www.feastingathome.com/vegan-fajitas/"
	label="Vegan Fajitas"
	alt="test" />)
	}
test('<Wrapper /> renders', async () => {
	const { container } = render(<Wrapper children={children}/>)
	expect(container).toMatchSnapshot()
})
test('<Wrapper /> cards exist', async () => {
	render(<Wrapper children={children}/>)
	const labels = await screen.getAllByAltText("test")
	expect(labels.length).toBe(10)
})
