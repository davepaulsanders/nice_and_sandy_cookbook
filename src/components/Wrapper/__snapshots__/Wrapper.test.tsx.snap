// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Wrapper /> renders 1`] = `
<div>
  <div
    class="grid gap-8 grid-cols-12 grid-rows-auto w-11/12 sm:mx-auto"
  >
    <div
      class="col-span-12 sm:col-span-3"
    >
      <div
        class="flex flex-col justify-content-center \\   items-center min-w-[20rem] max-w-[30rem] min-h-[20rem] shadow-lg rounded-md"
      >
        <div
          class="w-11/12"
        >
          <img
            alt="test"
            class="h-[27rem] w-full object-cover"
            src="https://www.feastingathome.com/wp-content/uploads/2024/10/Vegan-Fajitas-13.jpg"
          />
        </div>
        <div
          class="m-4"
        >
          <a
            class="text-xl hover:underline hover:cursor-pointer"
            href="https://www.feastingathome.com/vegan-fajitas/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Vegan Fajitas
          </a>
        </div>
      </div>
    </div>
    <div
      class="col-span-12 sm:col-span-3"
    >
      <div
        class="flex flex-col justify-content-center \\   items-center min-w-[20rem] max-w-[30rem] min-h-[20rem] shadow-lg rounded-md"
      >
        <div
          class="w-11/12"
        >
          <img
            alt="test"
            class="h-[27rem] w-full object-cover"
            src="https://www.feastingathome.com/wp-content/uploads/2024/10/Vegan-Fajitas-13.jpg"
          />
        </div>
        <div
          class="m-4"
        >
          <a
            class="text-xl hover:underline hover:cursor-pointer"
            href="https://www.feastingathome.com/vegan-fajitas/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Vegan Fajitas
          </a>
        </div>
      </div>
    </div>
    <div
      class="col-span-12 sm:col-span-3"
    >
      <div
        class="flex flex-col justify-content-center \\   items-center min-w-[20rem] max-w-[30rem] min-h-[20rem] shadow-lg rounded-md"
      >
        <div
          class="w-11/12"
        >
          <img
            alt="test"
            class="h-[27rem] w-full object-cover"
            src="https://www.feastingathome.com/wp-content/uploads/2024/10/Vegan-Fajitas-13.jpg"
          />
        </div>
        <div
          class="m-4"
        >
          <a
            class="text-xl hover:underline hover:cursor-pointer"
            href="https://www.feastingathome.com/vegan-fajitas/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Vegan Fajitas
          </a>
        </div>
      </div>
    </div>
    <div
      class="col-span-12 sm:col-span-3"
    >
      <div
        class="flex flex-col justify-content-center \\   items-center min-w-[20rem] max-w-[30rem] min-h-[20rem] shadow-lg rounded-md"
      >
        <div
          class="w-11/12"
        >
          <img
            alt="test"
            class="h-[27rem] w-full object-cover"
            src="https://www.feastingathome.com/wp-content/uploads/2024/10/Vegan-Fajitas-13.jpg"
          />
        </div>
        <div
          class="m-4"
        >
          <a
            class="text-xl hover:underline hover:cursor-pointer"
            href="https://www.feastingathome.com/vegan-fajitas/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Vegan Fajitas
          </a>
        </div>
      </div>
    </div>
    <div
      class="col-span-12 sm:col-span-3"
    >
      <div
        class="flex flex-col justify-content-center \\   items-center min-w-[20rem] max-w-[30rem] min-h-[20rem] shadow-lg rounded-md"
      >
        <div
          class="w-11/12"
        >
          <img
            alt="test"
            class="h-[27rem] w-full object-cover"
            src="https://www.feastingathome.com/wp-content/uploads/2024/10/Vegan-Fajitas-13.jpg"
          />
        </div>
        <div
          class="m-4"
        >
          <a
            class="text-xl hover:underline hover:cursor-pointer"
            href="https://www.feastingathome.com/vegan-fajitas/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Vegan Fajitas
          </a>
        </div>
      </div>
    </div>
    <div
      class="col-span-12 sm:col-span-3"
    >
      <div
        class="flex flex-col justify-content-center \\   items-center min-w-[20rem] max-w-[30rem] min-h-[20rem] shadow-lg rounded-md"
      >
        <div
          class="w-11/12"
        >
          <img
            alt="test"
            class="h-[27rem] w-full object-cover"
            src="https://www.feastingathome.com/wp-content/uploads/2024/10/Vegan-Fajitas-13.jpg"
          />
        </div>
        <div
          class="m-4"
        >
          <a
            class="text-xl hover:underline hover:cursor-pointer"
            href="https://www.feastingathome.com/vegan-fajitas/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Vegan Fajitas
          </a>
        </div>
      </div>
    </div>
    <div
      class="col-span-12 sm:col-span-3"
    >
      <div
        class="flex flex-col justify-content-center \\   items-center min-w-[20rem] max-w-[30rem] min-h-[20rem] shadow-lg rounded-md"
      >
        <div
          class="w-11/12"
        >
          <img
            alt="test"
            class="h-[27rem] w-full object-cover"
            src="https://www.feastingathome.com/wp-content/uploads/2024/10/Vegan-Fajitas-13.jpg"
          />
        </div>
        <div
          class="m-4"
        >
          <a
            class="text-xl hover:underline hover:cursor-pointer"
            href="https://www.feastingathome.com/vegan-fajitas/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Vegan Fajitas
          </a>
        </div>
      </div>
    </div>
    <div
      class="col-span-12 sm:col-span-3"
    >
      <div
        class="flex flex-col justify-content-center \\   items-center min-w-[20rem] max-w-[30rem] min-h-[20rem] shadow-lg rounded-md"
      >
        <div
          class="w-11/12"
        >
          <img
            alt="test"
            class="h-[27rem] w-full object-cover"
            src="https://www.feastingathome.com/wp-content/uploads/2024/10/Vegan-Fajitas-13.jpg"
          />
        </div>
        <div
          class="m-4"
        >
          <a
            class="text-xl hover:underline hover:cursor-pointer"
            href="https://www.feastingathome.com/vegan-fajitas/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Vegan Fajitas
          </a>
        </div>
      </div>
    </div>
    <div
      class="col-span-12 sm:col-span-3"
    >
      <div
        class="flex flex-col justify-content-center \\   items-center min-w-[20rem] max-w-[30rem] min-h-[20rem] shadow-lg rounded-md"
      >
        <div
          class="w-11/12"
        >
          <img
            alt="test"
            class="h-[27rem] w-full object-cover"
            src="https://www.feastingathome.com/wp-content/uploads/2024/10/Vegan-Fajitas-13.jpg"
          />
        </div>
        <div
          class="m-4"
        >
          <a
            class="text-xl hover:underline hover:cursor-pointer"
            href="https://www.feastingathome.com/vegan-fajitas/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Vegan Fajitas
          </a>
        </div>
      </div>
    </div>
    <div
      class="col-span-12 sm:col-span-3"
    >
      <div
        class="flex flex-col justify-content-center \\   items-center min-w-[20rem] max-w-[30rem] min-h-[20rem] shadow-lg rounded-md"
      >
        <div
          class="w-11/12"
        >
          <img
            alt="test"
            class="h-[27rem] w-full object-cover"
            src="https://www.feastingathome.com/wp-content/uploads/2024/10/Vegan-Fajitas-13.jpg"
          />
        </div>
        <div
          class="m-4"
        >
          <a
            class="text-xl hover:underline hover:cursor-pointer"
            href="https://www.feastingathome.com/vegan-fajitas/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Vegan Fajitas
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
